tensorflow_datasets/image_classification/caltech256/caltech256.py

```diff
@@ -0,0 +1,111 @@
+"""caltech256 dataset."""
+
+import os
+import numpy as np
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_DESCRIPTION = """
+Caltech-256 is an object recognition dataset spanning 257 classes (256 object classes and an additional clutter class).
+This dataset contains 30,607 real-world images, of different sizes, with each class having at least 80 images.
+Object categories are extremely diverse, ranging from grasshopper to tuning fork.
+This dataset is an improvement to its predecessor, the Caltech 101 dataset.
+"""
+
+_CITATION = """\
+@article{article,
+title = {Caltech-256 Object Category Dataset},
+author = {Griffin, Gregory and Holub, Alex and Perona, Pietro},
+journal = {CalTech Report}
+year = {2007},
+}
+"""
+
+_LABELS_FNAME = 'image_classification/caltech256/caltech256_labels.txt'
+
+_BASE_URL = 'https://drive.google.com/u/1/uc?id=1r6o0pSROcV1_VwT4oSjA2FBUSCWGuxLK&export=download'
+
+_TRAIN_POINTS_PER_CLASS = 75
+
+
+class Caltech256(tfds.core.GeneratorBasedBuilder):
+  """DatasetBuilder for caltech256 dataset."""
+
+  VERSION = tfds.core.Version('1.0.0')
+  RELEASE_NOTES = {
+    '1.0.0': 'Initial release.',
+  }
+
+  def _info(self) -> tfds.core.DatasetInfo:
+    """Returns the dataset metadata."""
+
+    class_labels_file = tfds.core.tfds_path(_LABELS_FNAME)
+    return tfds.core.DatasetInfo(
+      builder=self,
+      description=_DESCRIPTION,
+      features=tfds.features.FeaturesDict({
+        'image': tfds.features.Image(),
+        'label': tfds.features.ClassLabel(names_file=class_labels_file),
+        'image/file_name': tfds.features.Text(),
+      }),
+      supervised_keys=('image', 'label'),
+      # Homepage of the dataset
+      homepage='http://www.vision.caltech.edu/Image_Datasets/Caltech256/',
+      citation=_CITATION
+    )
+
+  def _split_generators(self, dl_manager: tfds.download.DownloadManager):
+    """Returns SplitGenerators."""
+    path = dl_manager.download_and_extract(_BASE_URL)
+    return [
+      tfds.core.SplitGenerator(
+        name=tfds.Split.TRAIN,
+        gen_kwargs={
+          'images_dir_path': path,
+          'is_train_split': True,
+        }),
+      tfds.core.SplitGenerator(
+        name=tfds.Split.TEST,
+        gen_kwargs={
+          'images_dir_path': path,
+          'is_train_split': False,
+        }),
+    ]
```

Comment:
Please follow the new convention
```
    return {
        'train': self._generate_examples(path=extracted_path / 'train_images'),
        'test': self._generate_examples(path=extracted_path / 'test_images'),
    }
```

Reply:
@vijayphoenix 
Modified the code as per new convention.